### PR TITLE
Retry requests to GitHub.com API

### PR DESCRIPTION
**Pull request by jugglinmike**

### CODE REVIEW

**Comment by jugglinmike** [comment]
We have logs for the past 7 days. Over that time, we saw three 404's from GitHub.com:

    $ grep 'Not Found' -B2 all.tsv | cut -f 2,10
    2018-10-22T06:44:07     GET https://api.github.com/repos/web-platform-tests/wpt/pulls/13645/files 
    2018-10-22T06:44:07     #13645: THIS SHOULDN'T EVER HAPPEN 
    2018-10-22T06:44:07      Error: Not Found 
    --
    2018-10-22T06:54:46     GET https://api.github.com/repos/web-platform-tests/wpt/issues/13646/events 
    2018-10-22T06:54:46     #13646: THIS SHOULDN'T EVER HAPPEN 
    2018-10-22T06:54:46      Error: Not Found 
    --
    2018-10-24T01:06:18     GET https://api.github.com/repos/web-platform-tests/wpt/pulls/13693/files 
    2018-10-24T01:06:18     #13693: THIS SHOULDN'T EVER HAPPEN 
    2018-10-24T01:06:18      Error: Not Found 

All of these happened to be automated pull requests, but I don't think that's relevant.

**Comment by zcorpan** [comment]
@jugglinmike should we merge this?

**Comment by jugglinmike** [comment]
That'd be great, but it needs to be rebased first

**Comment by zcorpan** [comment]
Rebased. The test in CI never completed but retried the 404 over and over, so I canceled the checks.

File: test/retry.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        });
+    });
+
+    test('retry until maxAttempts', function() {
+        var callCount = 0;
+        var action = function() { callCount += 1; };
+        var when = function() { return true; };
+
+        return retry({
+            maxAttempts: 23,
```
Comment (zcorpan) [inline test/retry.js]: Hmm, maybe the tests just have a large number of retries? Do we need this many retries?

**Comment by foolip** [comment]
I recently added https://github.com/octokit/plugin-throttling.js to https://github.com/w3c/validate-repos, and think something like that might be a good option to continuing to develop our own custom GitHub API bindings in this repo.

@stephenmcgruer?

**Comment by stephenmcgruer** [comment]
@foolip so I'm clear; your suggestion is to integrate https://github.com/octokit/plugin-throttling.js with this repo rather than merge this PR?

**Comment by foolip** [comment]
Yes, using  either https://github.com/octokit/rest.js or https://github.com/octokit/request.js together with that rate throttling plugins seems like it'd be less code to maintain over time. The difference between rest.js and request.js is that the former wraps everything in JS functions like `octokit.pulls.list(...)` while the latter is just `request("GET /repos/:owner/:repo/pulls", ...)`, closer to what this repo already does.

I've used [core.js](https://github.com/octokit/core.js)+throttling successfully in https://github.com/w3c/validate-repos/pull/79 so I know that works, but it turns out core.js depends on request.js and I think just request.js might do it for this project.

**Comment by foolip** [comment]
I just checked, it has to be https://github.com/octokit/core.js because https://github.com/octokit/request.js doesn't support plugins.

**Pull request closed without merging**